src/parthenon_manager.cpp

```diff
@@ -235,7 +235,7 @@ void ParthenonManager::RestartPackages(Mesh &rm, RestartReader &resfile) {
     // Being extra stringent here so that we don't forget to update the machinery when
     // another change happens.
     PARTHENON_REQUIRE_THROWS(
-        HDF5::OUTPUT_VERSION_FORMAT == 2,
+        HDF5::OUTPUT_VERSION_FORMAT == 3,
```

Comment:
Given @forrestglines's demonstration that for AthenaPK the HDF5 format is unchanged, I think we can think of this PR more as a bugfix for higher-than-vector-dimensional data. In that case, does it still make sense to provide a conversion script to go from file format `2` to `3`? The file would either be unchanged, or higher-than-vector data would be would be rewritten ambiguously since that is basically the issue this PR is fixing. Instead of such a conversion script is it ok to simply have this check be 
```c++
PARTHENON_REQUIRE_THROWS(HDF5::OUTPUT_VERSION_FORMAT == 2 || HDF5::OUTPUT_VERSION_FORMAT == 3, "Auto conversion from old to current format not implemented yet.");
if (HDF5::OUTPUT_VERSION_FORMAT == 2) {
PARTHENON_WARN("HDF5 output version format 2 may be erroneous for data other than cell-centered scalar and vector fields");
}
```
where I've added a warning for higher-than-vector-dimensional data?

Reply:
My intent was to point out that I believe the format is only the same if you don't use scalar variables, which KHARMA does.  We also have a lot of data in the old format (and actually quite a bit in the format prior to that one, too...)

However, the standard tool for plotting and converting KHARMA output just supports both formats.  So, it's no bother to me or (afaik) KHARMA users if the format changes without any printed warnings or Parthenon-provided migration tools.

Reply:
Currently this PR can restart from either its own dump format `3`, or the previous format `2` assuming only scalar and vector data. I personally don't see a need for additional fixes. @bprather if you think we could make additional changes here to improve your workflow let me know, but my impression is this PR as is should work pretty well for you.